proxy/sources/reddit.py

```
@@ -0,0 +1,157 @@
+import json
+import re
+import random
+from datetime import datetime
+
+from django.conf import settings
+from django.shortcuts import redirect
+from django.urls import re_path
+
+from ..source import ProxySource
+from ..source.data import ChapterAPI, SeriesAPI, SeriesPage
+from ..source.helpers import api_cache, get_wrapper
+
+class Reddit(ProxySource):
+    def get_reader_prefix(self):
+        return "reddit"
+
+    def shortcut_instantiator(self):
+        def handler(request, gallery_id):
+            return redirect(
+                f"reader-{self.get_reader_prefix()}-chapter-page",
+                gallery_id,
+                "1",
+                "1",
+            )
+
+        return [
+            re_path(r"^reddit/(?P<gallery_id>[\d\w]+)/$", handler),
+        ]
+
+    @staticmethod
+    def image_url_handler(url):
+        #transform thumbnail link from https://preview.redd.it/media_id.ext?junk
+        #                           to https://i.redd.it/media_id.ext
+        return re.sub(r"\?.*", "", url.replace("preview.redd.it", "i.redd.it"))
+
+
+    def reddit_api(self, meta_id):
+        resp = get_wrapper(f"https://api.reddit.com/api/info/?id=t3_{meta_id}&raw_json=1")
+        if resp.status_code != 200:
+            return None
+
+        api_data = json.loads(resp.text)
+        try: 
+            api_data = api_data["data"]["children"][0]["data"]
+
+            if not api_data["is_gallery"] or api_data["removed_by_category"] != None:
```

It doesn't look like `is_gallery` is guaranteed to be in the schema, but please correct me if I'm wrong. For reference, I took a peek at the [API response](https://api.reddit.com/api/info/?id=t3_q355c1&raw_json=1) to [this post](https://www.reddit.com/r/manga/comments/q355c1/disc_kaguyasama_wants_to_be_confessed_to_chapter/).

I noticed this entire block is wrapped with a try-catch, but do you expect it to be thrown anywhere else? In practice, we should let errors propagate up instead of handling them in each proxy implementation, but there isn't currently any handler at the top-level to deal with it. I can add it after this PR is merged.

---

I've changed it so it'll throw if the response has an unexpected format and returns none otherwise (like for missing `is_gallery`).